net-mgmt/pfSense-pkg-lldpd/files/usr/local/pkg/lldpd/lldpd.inc

```diff
@@ -120,11 +120,21 @@ function lldpd_sync_config() {
 	
 	/* Interface list */
 	$interfaces = array();
+	$platform = system_identify_specific_platform();
+	$int_switch = array('SG-1100', 'SG-3100');
```

Comment:
There are more platforms that just these with integrated switches. You should check for the presence of the function `switch_get_devices()`, if it's found, and has a non-empty result, then you know the device has a switch.

Might need to also check for the presence of `/etc/inc/switch.inc` and `require_once('switch.inc');` before that function is available, but I think it gets included early enough in other files that may not be necessary.

Reply:
got it! fixed

tested on SG-1100, XG-7100 and KVM